{"metadata":{"kernelspec":{"display_name":"Python 3 (ipykernel)","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.8.12"}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"markdown","source":"# Investigating evolutionary relationships among histone proteins through sequence and cluster analysis\n\nScientists often deal with large datasets in molecular genetics. For example, we might go online and run a BLAST search to identify a protein sequence. The default BLAST parameters are designed to limit the amount of data we receive to 100 sequences. In many cases this is more than sufficient, and we can stay out of \"Big Data\" weeds so to speak. \n\nHowever, protein (and nucleotide) sequences are rich with information, and there is much knowledge to be gained about evolutionary relationships between proteins using Big Data. Processing large datasets often requires specialized software packages. \n\nIn this project, we will: \n\n    1. Be able to handle, filter, and clean large sequence datasets\n    2. Learn about and use the basic local alignment sequence tool (BLAST)\n    3. Be able to create interactive sequence similarity networks from BLAST results to investigate sequence relationships\n    4. Be able to generate new knowledge using sequence similarity networks \n","metadata":{}},{"cell_type":"markdown","source":"## A (very) brief introduction to Jupyter Notebooks, Jupyter Lab, and Python\n\n\nJupyter notebooks provide an interactive, web-based programming environment. This format provides an easy way to access a variety of Big Data programming tools with instructions (like this section) and coding cells (in which you enter code). ","metadata":{}},{"cell_type":"markdown","source":"This notebook is based on the Python 3 programming language. Let's type some Python 3 code in the coding cell below:\n\n<font color=blue><b>STEP 1:</b></font> Type in the code provided by Dr. Schurko. While the code is running you will see an asterisk appear in the brackets, e.g.: [\\*]. When it is completed, a number appears, e.g. [1].","metadata":{}},{"cell_type":"code","source":"","metadata":{"trusted":true},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"<b>Congratulations!<b>\n\n<font color=blue><b>STEP 2:</b></font>Let's add to that by adding your name to the end of the statement. You can simply re-run the code in the coding cell above. When you do, you will see the new output below that box and the number in brackets on that line increments by 1.\n\nIf this is your first time writing and running some Python code, congratulations! ","metadata":{}},{"cell_type":"markdown","source":"<font color=blue><b>STEP 3:</b></font> Let's edit that code further to make it print out that statement three times. This code will create a loop and run it over the range given.\n","metadata":{}},{"cell_type":"markdown","source":"<font color=blue><b>STEP 4:</b></font> Before we move on, change the range in the code above to 10 and run it again. Even more awesomeness!\n***","metadata":{}},{"cell_type":"markdown","source":"<font color=blue><b>STEP 5:</b></font> Now, let's do some calcuations. In the coding cell below, type in a calculation as discussed by Dr. Schurko:","metadata":{}},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"<font color=blue><b>STEP 6:</b></font> To make better use, we can have Ptyhon do calculations for us. We will assign some variables to calculate how to dilute a stock concentration to a working concentration using the C1V1 = C2V2 formula. In the cell below, assign variables for C1, V1 and C2. We will then solve for V2:","metadata":{}},{"cell_type":"code","source":"","metadata":{"trusted":true},"execution_count":8,"outputs":[{"name":"stdout","text":"0.01\n","output_type":"stream"}]},{"cell_type":"markdown","source":"<font color=blue><b>STEP 8:</b></font> Great! Now, in the space below let's put our solution for V2 in the form of a sentence:","metadata":{"tags":[]}},{"cell_type":"code","source":"","metadata":{"trusted":true},"execution_count":9,"outputs":[{"name":"stdout","text":"The volume you need to add is 0.01\n","output_type":"stream"}]},{"cell_type":"markdown","source":"From these simple exercises, we now have a basic idea for how to run working code with minimal guidance.\n\n\n\n<b>Now onto today's project<b>\n\nIn this exercise we will use the histone protein sequence files that we downloaded last week from UniProt. You should have one text file that contains all of the histone sequences in FASTA format (there should be >8400 FASTA sequences in your file). Today, we will utilize NCBI BLAST software to use E-values for visualization, the CD-HIT software to reduce our data set size in a rational way, and cytoscape for viewing sequence similarity networks.\n\nWhile it is possible to install most of this software on any computer, it can be very difficult to get everything working. This is why we are running this Jupyter notebook in Binder, a <i> remote </i> environment that I have set up. In this environment all the needed software is already installed, and we can easily upload files to work with! \n    \n## Some important ideas for using Jupyter Notebooks in Binder\n\n    1. Please ensure the code in a cell has finished running before moving on to the next one!\n    2. If you accidentally delete or mangle some code, use the Edit -> Undo function!\n    3. If you need to stop some code from running in a cell use the Stop button (next to the Run button).\n    4. If you don't interact with Binder for too long, you may lose the Kernel and need to restart it. \n    5. If you are away for a long time, then you may need to relaunch your Binder - which means you might need to regenerate any data that you did not download!\n***\n","metadata":{}},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"!clustalo -i <<your_file_name>> --outfmt='msf' -o files/8_histone.msf","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"<font color=blue><b>STEP 5:</b></font> Find the 8_histones.msf file in the files tab and double click on it to open it. The symbol '~' indicates that a particular sequence does not have additional amino acids at the N or C terminus. A '.' means there is no amino acid in that sequence at that position, also called a gap. \n\n<font color=blue><b>STEP 6:</b></font> Use the msf to answer the following questions:\n\n    This should be similar to the result we observed last week. What is different about the histone H2A sequence from the rotifer Adineta vaga?\n\n\n<font color=blue><b>STEP 7:</b></font> When you are done, open Notebook \"2 - Sequences_and_BLAST\" by double clicking on the file in the left hand panel. The code cells in 2 - Sequences_and_BLAST already have the needed code, to keep you engaged and on your toes, you will have to edit that code in specific (and hopefully clearly annotated) ways.\n\nTEST\n","metadata":{}},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]}]}