{"metadata":{"kernelspec":{"name":"python3","display_name":"Python 3 (ipykernel)","language":"python"},"language_info":{"name":"python","version":"3.8.12","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"markdown","source":"# Investigating evolutionary relationships of histone proteins through sequence and cluster analysis!\n# On to the Science - Importing some sequence data!\n\n<font color=blue><b>STEP 1:</b></font> Upload your file containing histone sequences named <b>uniprot_histone.fasta</b> to this Binder envrionment. In the files panel at the left, double click the files folder to open it. You should see one existing file in there named <b>known_histone.fasta</b>.\n    \nEither drag the <b>uniprot_histone.fasta</b> file from your computer into the files folder or click the Upload button in the left panel. When done, you should have two files in your files folder.\n\nYou can double click it on either file in your files folder and it will display the contents in a new window.","metadata":{}},{"cell_type":"markdown","source":"# Working with the sequences in BioPython\n\nThe code boxes below will increase in complexity as we go on. Comments in the code begin with #s. Read these if you want help understanding the code.\n\nIn the first code box below, the first line \"turns on\" the SeqIO function of Biopython, a package (set of tools) built for biology and biochemistry! You can learn more at https://biopython.org/\n\nThe second line uses SeqIO (think: sequence input and output) to read a fasta file and stores the information as a list of records.\n\n<font color=blue><b>STEP 2:</b></font> The command below won't run correctly unless you enter in the file to read. <b> Between the quotes, where it says <<<your file here>>>, change it to : files/uniprot_histone.fasta.</b>\n    ","metadata":{}},{"cell_type":"code","source":"from Bio import SeqIO # imports the SeqIO function from Biopython\n\nrecords = list(SeqIO.parse(\"<<<your file here>>>\", \"fasta\"))     # reads the fasta file into a list of records \nprint(\"Finished storing the FASTA file in the list called \\\"records\\\".\")","metadata":{},"execution_count":null,"outputs":[{"name":"stdout","text":"Finished storing the FASTA file in the list called \"records\".\n","output_type":"stream"}]},{"cell_type":"markdown","source":"Now, let's see how we can access different information about the sequence records stored in the list of records.\n\n<font color=blue><b>STEP 3:</b></font> The first line below will tell us how many sequences were in the FASTA file, and then the information stored in the first item. Note that python numbering starts with the number 0. Go ahead and run the code box below.","metadata":{}},{"cell_type":"code","source":"print(\"There are %i sequences in your file.\\n\" % len(records))   # prints the number of sequences, that is, the length of the list, named records\n\nprint(records[0]) #prints the information in the first record","metadata":{"tags":[],"collapsed":true,"jupyter":{"outputs_hidden":true}},"execution_count":null,"outputs":[{"name":"stdout","text":"There are 8474 sequences in your file.\n\nID: sp|P68431|H31_HUMAN\nName: sp|P68431|H31_HUMAN\nDescription: sp|P68431|H31_HUMAN Histone H3.1 OS=Homo sapiens OX=9606 GN=H3C1 PE=1 SV=2\nNumber of features: 0\nSeq('MARTKQTARKSTGGKAPRKQLATKAARKSAPATGGVKKPHRYRPGTVALREIRR...ERA')\n","output_type":"stream"}]},{"cell_type":"markdown","source":"You should be able to see that first record includes several things: an ID, Name, Description, and a Sequence.\n\nWe can access each of those items specifically\n\n<font color=blue><b>STEP 4:</b></font>  Run the code below to print only the sequence of the first record.","metadata":{"tags":[]}},{"cell_type":"code","source":"print(records[0].seq)","metadata":{},"execution_count":null,"outputs":[{"name":"stdout","text":"MARTKQTARKSTGGKAPRKQLATKAARKSAPATGGVKKPHRYRPGTVALREIRRYQKSTELLIRKLPFQRLVREIAQDFKTDLRFQSSAVMALQEACEAYLVGLFEDTNLCAIHAKRVTIMPKDIQLARRIRGERA\n","output_type":"stream"}]},{"cell_type":"markdown","source":"The next code box shows us how to list the first 10 ids. Then it lists the first record id and its sequence.\n\n<font color=blue><b>STEP 5:</b></font> Run the code box below.","metadata":{}},{"cell_type":"code","source":"print(\"The first 10 sequence record ids are:\\n\")\nfor i in range(10):                                              # this creates a variable i and counts to 10\n    print(records[i].id)                                         # prints the id for record i\n    \nprint(\"\\nThe record: %s has a sequence of: %s\\n\" % (records[-1].id, records[-1].seq))  # prints the record id and its sequence!","metadata":{},"execution_count":null,"outputs":[{"name":"stdout","text":"The first 10 sequence record ids are:\n\nsp|P68431|H31_HUMAN\nsp|P84228|H32_MOUSE\nsp|P62807|H2B1C_HUMAN\nsp|Q71DI3|H32_HUMAN\nsp|Q96EQ9|PRDM9_MOUSE\nsp|Q9NQV7|PRDM9_HUMAN\nsp|P0C6Y7|PRDM9_RAT\nsp|Q6P2A1|PRDM9_DANRE\nsp|Q6FI13|H2A2A_HUMAN\nsp|Q92830|KAT2A_HUMAN\n\nThe record: sp|Q9WTN3|SRBP1_MOUSE has a sequence of: MDELAFGEAALEQTLAEMCELDTAVLNDIEDMLQLINNQDSDFPGLFDAPYAGGETGDTGPSSPGANSPESFSSASLASSLEAFLGGPKVTPAPLSPPPSAPAALKMYPSVSPFSPGPGIKEEPVPLTILQPAAPQPSPGTLLPPSFPAPPVQLSPAPVLGYSSLPSGFSGTLPGNTQQPPSSLPLAPAPGVLPTPALHTQVQSLASQQPLPASAAPRTNTVTSQVQQVPVVLQPHFIKADSLLLTAVKTDAGATVKTAGISTLAPGTAVQAGPLQTLVSGGTILATVPLVVDTDKLPIHRLAAGSKALGSAQSRGEKRTAHNAIEKRYRSSINDKIVELKDLVVGTEAKLNKSAVLRKAIDYIRFLQHSNQKLKQENLTLRSAHKSKSLKDLVSACGSGGGTDVSMEGMKPEVVETLTPPPSDAGSPSQSSPLSFGSRASSSGGSDSEPDSPAFEDSQVKAQRLPSHSRGMLDRSRLALCVLAFLCLTCNPLASLFGWGILTPSDATGTHRSSGRSMLEAESRDGSNWTQWLLPPLVWLANGLLVLACLALLFVYGEPVTRPHSGPAVHFWRHRKQADLDLARGDFPQAAQQLWLALQALGRPLPTSNLDLACSLLWNLIRHLLQRLWVGRWLAGQAGGLLRDRGLRKDARASARDAAVVYHKLHQLHAMGKYTGGHLAASNLALSALNLAECAGDAISMATLAEIYVAAALRVKTSLPRALHFLTRFFLSSARQACLAQSGSVPLAMQWLCHPVGHRFFVDGDWAVHGAPPESLYSVAGNPVDPLAQVTRLFREHLLERALNCIAQPSPGAADGDREFSDALGYLQLLNSCSDAAGAPACSFSVSSSMAATTGPDPVAKWWASLTAVVIHWLRRDEEAAERLYPLVEHIPQVLQDTERPLPRAALYSFKAARALLDHRKVESSPASLAICEKASGYLRDSLASTPTGSSIDKAMQLLLCDLLLVARTSLWQRQQSPASVQVAHGTSNGPQASALELRGFQHDLSSLRRLAQSFRPAMRRVFLHEATARLMAGASPARTHQLLDRSLRRRAGSSGKGGTTAELEPRPTWREHTEALLLASCYLPPAFLSAPGQRMSMLAEAARTVEKLGDHRLLLDCQQMLLRLGGGTTVTSS\n\n","output_type":"stream"}]},{"cell_type":"markdown","source":"***\nGreat! The code above finds the first record (recall in Python we start counting at zero), so records[0].id gets the identification of the first record. \n\n<font color=blue><b>STEP 5:</b></font> Edit the last print statement in the code above to give the id and sequence of the 100th record.\n\nWe can also look at the last record id. You could put in the number of records (less one), but -1 is easier! The -1 starts counting from the end of the records list and you don't need to know how many records you have.\n\n<font color=blue><b>STEP 6:</b></font> Edit the last print statement in the code above to give the id and sequence of last record.\n\n***","metadata":{}},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"<font color=blue><b>STEP 7:</b></font> We can also search our sequences by some other search terms. Examine the comments below to see how we can do this. You will need to edit the line that sets the variable search_term. In the code box below, change the text <b>\\<<<search term here\\>>></b> to a search term of your own. Then run the code box.\n\n    Where to search ideas: id, description, name","metadata":{}},{"cell_type":"code","source":"import re  # imports the re (Regular Expressions) functions\n\nsearch_term = \"<<<search term here>>>\"   # in between the quotes we can add a search term.\n                                         \n\ncounter = 0   # a variable to keep track of how many times we find the term\n\nfor item in records:                       # this creates a loop to iterate over all the records               \n    if re.search(search_term, item.description, re.IGNORECASE):    # Here we use re.search to find the search term in the item ID, returns true if yes\n        print(item.description)                     # If the above is true we print the item ID...\n        counter = counter + 1              # and increment our counter\n    else: continue                         # if the search term wasn't found we do nothing, just continue on\n\n        \n# the next two lines print a summary for us - either no hits or how many hits.\n        \nif not counter: print(\"We didn't find any results using the search term: %s\" % search_term)\nelse: print(\"\\nWe found %i records using the search term: %s\" % (counter, search_term))","metadata":{},"execution_count":null,"outputs":[{"name":"stdout","text":"sp|Q6FI13|H2A2A_HUMAN Histone H2A type 2-A OS=Homo sapiens OX=9606 GN=H2AC18 PE=1 SV=3\nsp|P04908|H2A1B_HUMAN Histone H2A type 1-B/E OS=Homo sapiens OX=9606 GN=H2AC4 PE=1 SV=2\nsp|Q16777|H2A2C_HUMAN Histone H2A type 2-C OS=Homo sapiens OX=9606 GN=H2AC20 PE=1 SV=4\nsp|Q9P0M6|H2AW_HUMAN Core histone macro-H2A.2 OS=Homo sapiens OX=9606 GN=MACROH2A2 PE=1 SV=3\nsp|Q6FWW0|CHZ1_CANGA Histone H2A.Z-specific chaperone CHZ1 OS=Candida glabrata (strain ATCC 2001 / CBS 138 / JCM 3761 / NBRC 0622 / NRRL Y-65) OX=284593 GN=CHZ1 PE=3 SV=1\nsp|Q4WWC6|H2A_ASPFU Histone H2A OS=Neosartorya fumigata (strain ATCC MYA-4609 / Af293 / CBS 101355 / FGSC A1100) OX=330879 GN=hta1 PE=3 SV=3\nsp|P40282|H2A_PLAFA Histone H2A OS=Plasmodium falciparum OX=5833 PE=2 SV=1\nsp|Q9M531|H2A_EUPES Histone H2A OS=Euphorbia esula OX=3993 PE=2 SV=1\nsp|Q2HH37|H2A_CHAGB Histone H2A OS=Chaetomium globosum (strain ATCC 6205 / CBS 148.51 / DSM 1962 / NBRC 6347 / NRRL 1970) OX=306901 GN=HTA1 PE=3 SV=2\nsp|Q4HTT1|H2A_GIBZE Histone H2A OS=Gibberella zeae (strain ATCC MYA-4620 / CBS 123657 / FGSC 9075 / NRRL 31084 / PH-1) OX=229533 GN=HTA1 PE=3 SV=3\nsp|P02264|H2A_ONCMY Histone H2A OS=Oncorhynchus mykiss OX=8022 PE=1 SV=2\nsp|Q5M8Q2|H2AL1_MOUSE Histone H2A-like 1 OS=Mus musculus OX=10090 GN=H2al1a PE=2 SV=1\nsp|P02274|H2A1_TETPY Histone H2A.1 OS=Tetrahymena pyriformis OX=5908 GN=HTA2 PE=1 SV=2\nsp|P0C5Z0|H2AB2_HUMAN Histone H2A-Bbd type 2/3 OS=Homo sapiens OX=9606 GN=H2AB2 PE=1 SV=1\nsp|P16104|H2AX_HUMAN Histone H2AX OS=Homo sapiens OX=9606 GN=H2AX PE=1 SV=2\nsp|Q2UBH8|CHZ1_ASPOR Histone H2A.Z-specific chaperone chz1 OS=Aspergillus oryzae (strain ATCC 42149 / RIB 40) OX=510516 GN=chz1 PE=3 SV=1\nsp|Q0C807|CHZ1_ASPTN Histone H2A.Z-specific chaperone chz1 OS=Aspergillus terreus (strain NIH 2624 / FGSC A1156) OX=341663 GN=chz1 PE=3 SV=1\nsp|A6SMQ7|CHZ1_BOTFB Histone H2A.Z-specific chaperone chz1 OS=Botryotinia fuckeliana (strain B05.10) OX=332648 GN=chz1 PE=3 SV=1\nsp|Q2GNS7|CHZ1_CHAGB Histone H2A.Z-specific chaperone CHZ1 OS=Chaetomium globosum (strain ATCC 6205 / CBS 148.51 / DSM 1962 / NBRC 6347 / NRRL 1970) OX=306901 GN=CHZ1 PE=3 SV=2\nsp|A8N7N4|CHZ1_COPC7 Histone H2A.Z-specific chaperone CHZ1 OS=Coprinopsis cinerea (strain Okayama-7 / 130 / ATCC MYA-4618 / FGSC 9003) OX=240176 GN=CHZ1 PE=3 SV=1\nsp|Q5B2V2|CHZ1_EMENI Histone H2A.Z-specific chaperone chz1 OS=Emericella nidulans (strain FGSC A4 / ATCC 38163 / CBS 112.46 / NRRL 194 / M139) OX=227321 GN=chz1 PE=3 SV=1\nsp|Q6CWY0|CHZ1_KLULA Histone H2A.Z-specific chaperone CHZ1 OS=Kluyveromyces lactis (strain ATCC 8585 / CBS 2359 / DSM 70799 / NBRC 1267 / NRRL Y-1140 / WM37) OX=284590 GN=CHZ1 PE=3 SV=1\nsp|Q59RN6|CHZ1_CANAL Histone H2A.Z-specific chaperone CHZ1 OS=Candida albicans (strain SC5314 / ATCC MYA-2876) OX=237561 GN=CHZ1 PE=3 SV=2\nsp|Q1E4L8|CHZ1_COCIM Histone H2A.Z-specific chaperone CHZ1 OS=Coccidioides immitis (strain RS) OX=246410 GN=CHZ1 PE=3 SV=2\nsp|B0D1Q8|CHZ1_LACBS Histone H2A.Z-specific chaperone CHZ1 OS=Laccaria bicolor (strain S238N-H82 / ATCC MYA-4686) OX=486041 GN=CHZ1 PE=3 SV=1\nsp|A2Y5G8|H2A4_ORYSI Probable histone H2A.4 OS=Oryza sativa subsp. indica OX=39946 GN=OsI_019561 PE=3 SV=1\nsp|A2ZL69|H2AXB_ORYSI Probable histone H2AXb OS=Oryza sativa subsp. indica OX=39946 GN=OsI_037312 PE=3 SV=1\nsp|P70082|H2AJ_CHICK Histone H2A.J OS=Gallus gallus OX=9031 GN=H2A-IX PE=3 SV=1\nsp|A2YMC5|H2A1_ORYSI Probable histone H2A.1 OS=Oryza sativa subsp. indica OX=39946 GN=OsI_26380 PE=3 SV=2\nsp|P16866|H2A4_VOLCA Histone H2A-IV OS=Volvox carteri OX=3067 PE=3 SV=1\nsp|Q2HU65|H2A2_MEDTR Probable histone H2A.2 OS=Medicago truncatula OX=3880 GN=MtrDRAFT_AC149210g4v1 PE=3 SV=1\nsp|Q9BTM1|H2AJ_HUMAN Histone H2A.J OS=Homo sapiens OX=9606 GN=H2AJ PE=1 SV=1\nsp|Q9CQ70|H2AB1_MOUSE Histone H2A-Bbd type 1 OS=Mus musculus OX=10090 GN=H2ab1 PE=1 SV=1\nsp|P27661|H2AX_MOUSE Histone H2AX OS=Mus musculus OX=10090 GN=H2ax PE=1 SV=2\nsp|Q755T5|CHZ1_ASHGO Histone H2A.Z-specific chaperone CHZ1 OS=Ashbya gossypii (strain ATCC 10895 / CBS 109.51 / FGSC 9923 / NRRL Y-1056) OX=284811 GN=CHZ1 PE=3 SV=1\nsp|A1CQS5|CHZ1_ASPCL Histone H2A.Z-specific chaperone chz1 OS=Aspergillus clavatus (strain ATCC 1007 / CBS 513.65 / DSM 816 / NCTC 3887 / NRRL 1 / QM 1276 / 107) OX=344612 GN=chz1 PE=3 SV=2\nsp|Q4WJ31|CHZ1_ASPFU Histone H2A.Z-specific chaperone chz1 OS=Neosartorya fumigata (strain ATCC MYA-4609 / Af293 / CBS 101355 / FGSC A1100) OX=330879 GN=chz1 PE=3 SV=2\nsp|A5E145|CHZ1_LODEL Histone H2A.Z-specific chaperone CHZ1 OS=Lodderomyces elongisporus (strain ATCC 11503 / CBS 2605 / JCM 1781 / NBRC 1676 / NRRL YB-4239) OX=379508 GN=CHZ1 PE=3 SV=1\nsp|A4QYF9|CHZ1_MAGO7 Histone H2A.Z-specific chaperone CHZ1 OS=Magnaporthe oryzae (strain 70-15 / ATCC MYA-4617 / FGSC 8958) OX=242507 GN=CHZ1 PE=3 SV=1\nsp|A5DLG8|CHZ1_PICGU Histone H2A.Z-specific chaperone CHZ1 OS=Meyerozyma guilliermondii (strain ATCC 6260 / CBS 566 / DSM 6381 / JCM 1539 / NBRC 10279 / NRRL Y-324) OX=294746 GN=CHZ1 PE=3 SV=2\nsp|A3LXX5|CHZ1_PICST Histone H2A.Z-specific chaperone CHZ1 OS=Scheffersomyces stipitis (strain ATCC 58785 / CBS 6054 / NBRC 10063 / NRRL Y-11545) OX=322104 GN=CHZ1 PE=3 SV=1\nsp|Q10239|CHZ1_SCHPO Histone H2A.Z-specific chaperone chz1 OS=Schizosaccharomyces pombe (strain 972 / ATCC 24843) OX=284812 GN=chz1 PE=3 SV=1\nsp|A7ER98|CHZ1_SCLS1 Histone H2A.Z-specific chaperone chz1 OS=Sclerotinia sclerotiorum (strain ATCC 18683 / 1980 / Ss-1) OX=665079 GN=chz1 PE=3 SV=1\nsp|A6ZQX9|CHZ1_YEAS7 Histone H2A.Z-specific chaperone CHZ1 OS=Saccharomyces cerevisiae (strain YJM789) OX=307796 GN=CHZ1 PE=3 SV=2\nsp|Q84NJ4|H2A3_ORYSJ Probable histone H2A.3 OS=Oryza sativa subsp. japonica OX=39947 GN=Os08g0427700 PE=3 SV=1\nsp|P70094|H2AZL_XENLA Histone H2A.Z-like OS=Xenopus laevis OX=8355 GN=h2a.zl1 PE=2 SV=3\nsp|A5DQL2|H2AZ_PICGU Histone H2A.Z OS=Meyerozyma guilliermondii (strain ATCC 6260 / CBS 566 / DSM 6381 / JCM 1539 / NBRC 10279 / NRRL Y-324) OX=294746 GN=HTZ1 PE=3 SV=1\nsp|Q93077|H2A1C_HUMAN Histone H2A type 1-C OS=Homo sapiens OX=9606 GN=H2AC6 PE=1 SV=3\nsp|C0HKE2|H2A1C_MOUSE Histone H2A type 1-C OS=Mus musculus OX=10090 GN=H2ac6 PE=1 SV=1\nsp|P0C169|H2A1C_RAT Histone H2A type 1-C OS=Rattus norvegicus OX=10116 PE=1 SV=2\nsp|C0HKE3|H2A1D_MOUSE Histone H2A type 1-D OS=Mus musculus OX=10090 GN=H2ac7 PE=1 SV=1\nsp|O81826|H2A3_ARATH Probable histone H2A.3 OS=Arabidopsis thaliana OX=3702 GN=At4g27230 PE=1 SV=1\nsp|P02275|H2A1_WHEAT Histone H2A.1 OS=Triticum aestivum OX=4565 GN=H2A-9 PE=1 SV=2\nsp|Q99878|H2A1J_HUMAN Histone H2A type 1-J OS=Homo sapiens OX=9606 GN=H2AC14 PE=1 SV=3\nsp|P02262|H2A1_RAT Histone H2A type 1 OS=Rattus norvegicus OX=10116 PE=1 SV=2\nsp|A5DXS8|H2A2_LODEL Histone H2A.2 OS=Lodderomyces elongisporus (strain ATCC 11503 / CBS 2605 / JCM 1781 / NBRC 1676 / NRRL YB-4239) OX=379508 GN=HTA2 PE=3 SV=1\nsp|P35065|H2A1_TETTS Histone H2A.1 OS=Tetrahymena thermophila (strain SB210) OX=312017 GN=HTA2 PE=1 SV=2\nsp|Q64522|H2A2B_MOUSE Histone H2A type 2-B OS=Mus musculus OX=10090 GN=H2ac21 PE=1 SV=3\nsp|A5DBG4|H2A1_PICGU Histone H2A.1 OS=Meyerozyma guilliermondii (strain ATCC 6260 / CBS 566 / DSM 6381 / JCM 1539 / NBRC 10279 / NRRL Y-324) OX=294746 GN=HTA1 PE=3 SV=1\nsp|Q9T0H7|H2A8_ARATH Histone H2A.8 OS=Arabidopsis thaliana OX=3702 GN=At4g13570 PE=1 SV=1\nsp|Q43213|H2A5_WHEAT Protein H2A.5 OS=Triticum aestivum OX=4565 GN=H2A-2 PE=2 SV=3\nsp|C0HKE6|H2A1I_MOUSE Histone H2A type 1-I OS=Mus musculus OX=10090 GN=H2ac13 PE=1 SV=1\nsp|P02273|H2AX_TETPY Histone H2AX OS=Tetrahymena pyriformis OX=5908 GN=HTA1 PE=1 SV=2\nsp|P02263|H2A4_CHICK Histone H2A-IV OS=Gallus gallus OX=9031 PE=1 SV=2\nsp|Q32LA7|H2AV_BOVIN Histone H2A.V OS=Bos taurus OX=9913 GN=H2AZ2 PE=2 SV=3\nsp|Q27511|H2AV_CAEEL Histone H2A.V OS=Caenorhabditis elegans OX=6239 GN=htz-1 PE=1 SV=3\nsp|P02272|H2AV_CHICK Histone H2A.V OS=Gallus gallus OX=9031 GN=H2AZ2 PE=1 SV=2\nsp|O65759|H2AX_CICAR Histone H2AX OS=Cicer arietinum OX=3827 GN=HIS2A PE=2 SV=1\nsp|Q98SM3|EZH2A_XENLA Histone-lysine N-methyltransferase EZH2 OS=Xenopus laevis OX=8355 GN=ezh2-a PE=2 SV=1\nsp|P0C0S4|H2AZ_BOVIN Histone H2A.Z OS=Bos taurus OX=9913 GN=H2AZ1 PE=1 SV=2\nsp|O74268|H2A_BOTFB Histone H2A OS=Botryotinia fuckeliana (strain B05.10) OX=332648 GN=hta1 PE=3 SV=4\nsp|P0CO01|H2AZ_CRYNB Histone H2A.Z OS=Cryptococcus neoformans var. neoformans serotype D (strain B-3501A) OX=283643 GN=HTZ1 PE=3 SV=1\nsp|P0CN99|H2A_CRYNB Histone H2A OS=Cryptococcus neoformans var. neoformans serotype D (strain B-3501A) OX=283643 GN=HTA1 PE=3 SV=1\nsp|P0C952|H2A_ASPNG Histone H2A OS=Aspergillus niger OX=5061 GN=httA PE=2 SV=1\nsp|P09588|H2A_CAEEL Histone H2A OS=Caenorhabditis elegans OX=6239 GN=his-3 PE=1 SV=2\nsp|P40279|H2A_EUGGR Histone H2A OS=Euglena gracilis OX=3039 PE=2 SV=1\nsp|C0HJB7|H2A_EUPCP Buforin-EC OS=Euphlyctis cyanophlyctis OX=58519 PE=1 SV=1\nsp|P84052|H2A_DROER Histone H2A OS=Drosophila erecta OX=7220 GN=His2A PE=3 SV=2\nsp|Q875B8|H2A_PODAS Histone H2A OS=Podospora anserina OX=2587412 GN=HTA1 PE=3 SV=3\nsp|Q8I0T3|H2A_MYTED Histone H2A OS=Mytilus edulis OX=6550 PE=3 SV=3\nsp|Q0U1A1|H2A_PHANO Histone H2A OS=Phaeosphaeria nodorum (strain SN15 / ATCC MYA-4574 / FGSC 10173) OX=321614 GN=HTA1 PE=3 SV=1\nsp|Q59VP2|H2A2_CANAL Histone H2A.2 OS=Candida albicans (strain SC5314 / ATCC MYA-2876) OX=237561 GN=HTA2 PE=3 SV=3\nsp|P35062|H2A3_CHICK Histone H2A-III OS=Gallus gallus OX=9031 PE=1 SV=2\nsp|Q6FM31|H2A2_CANGA Histone H2A.2 OS=Candida glabrata (strain ATCC 2001 / CBS 138 / JCM 3761 / NBRC 0622 / NRRL Y-65) OX=284593 GN=HTA2 PE=3 SV=3\nsp|A9UMV8|H2AJ_RAT Histone H2A.J OS=Rattus norvegicus OX=10116 GN=H2aj PE=2 SV=1\nsp|P69142|H2AE_PSAMI Histone H2A, embryonic OS=Psammechinus miliaris OX=7660 PE=3 SV=2\nsp|P04736|H2A2_PSAMI Late histone H2A.2.1 OS=Psammechinus miliaris OX=7660 PE=2 SV=2\nsp|Q74ZL4|H2A1_ASHGO Histone H2A.1 OS=Ashbya gossypii (strain ATCC 10895 / CBS 109.51 / FGSC 9923 / NRRL Y-1056) OX=284811 GN=HTA1 PE=3 SV=3\nsp|Q8BFU2|H2A3_MOUSE Histone H2A type 3 OS=Mus musculus OX=10090 GN=H2aw PE=1 SV=3\nsp|Q8CCK0|H2AW_MOUSE Core histone macro-H2A.2 OS=Mus musculus OX=10090 GN=Macroh2a2 PE=1 SV=3\nsp|Q6BRG3|H2A2_DEBHA Histone H2A.2 OS=Debaryomyces hansenii (strain ATCC 36239 / CBS 767 / BCRC 21394 / JCM 1990 / NBRC 0083 / IGC 2968) OX=284592 GN=HTA2 PE=3 SV=3\nsp|P27892|H2A2_LEIIN Histone H2A.2 OS=Leishmania infantum OX=5671 PE=2 SV=1\nsp|Q9C681|H2A1_ARATH Probable histone H2A.1 OS=Arabidopsis thaliana OX=3702 GN=At1g51060 PE=1 SV=1\nsp|P04911|H2A1_YEAST Histone H2A.1 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=HTA1 PE=1 SV=2\nsp|A2YMC6|H2A2_ORYSI Probable histone H2A.2 OS=Oryza sativa subsp. indica OX=39946 GN=OsI_025469 PE=3 SV=1\nsp|Q94F49|H2A5_ARATH Probable histone H2A.5 OS=Arabidopsis thaliana OX=3702 GN=At5g27670 PE=1 SV=1\nsp|Q8R1M2|H2AJ_MOUSE Histone H2A.J OS=Mus musculus OX=10090 GN=H2aj PE=1 SV=1\nsp|A2QPM6|CHZ1_ASPNC Histone H2A.Z-specific chaperone chz1 OS=Aspergillus niger (strain CBS 513.88 / FGSC A1513) OX=425011 GN=chz1 PE=3 SV=1\nsp|A1D3N8|CHZ1_NEOFI Histone H2A.Z-specific chaperone chz1 OS=Neosartorya fischeri (strain ATCC 1020 / DSM 3700 / CBS 544.65 / FGSC A1164 / JCM 1740 / NRRL 181 / WB 181) OX=331117 GN=chz1 PE=3 SV=1\nsp|Q9P534|CHZ1_NEUCR Histone H2A.Z-specific chaperone chz-1 OS=Neurospora crassa (strain ATCC 24698 / 74-OR23-1A / CBS 708.71 / DSM 1257 / FGSC 987) OX=367110 GN=chz-1 PE=3 SV=2\nsp|Q0V3E0|CHZ1_PHANO Histone H2A.Z-specific chaperone CHZ1 OS=Phaeosphaeria nodorum (strain SN15 / ATCC MYA-4574 / FGSC 10173) OX=321614 GN=CHZ1 PE=3 SV=3\nsp|A7TR90|CHZ1_VANPO Histone H2A.Z-specific chaperone CHZ1 OS=Vanderwaltozyma polyspora (strain ATCC 22028 / DSM 70294 / BCRC 21397 / CBS 2163 / NBRC 10782 / NRRL Y-8283 / UCD 57-17) OX=436907 GN=CHZ1 PE=3 SV=1\nsp|P40019|CHZ1_YEAST Histone H2A.Z-specific chaperone CHZ1 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=CHZ1 PE=1 SV=2\nsp|Q9LZ46|H2A4_ARATH Probable histone H2A.4 OS=Arabidopsis thaliana OX=3702 GN=At5g02560 PE=1 SV=1\nsp|O93327|H2AY_CHICK Core histone macro-H2A.1 OS=Gallus gallus OX=9031 GN=MACROH2A1 PE=1 SV=3\nsp|Q2UJ80|H2AZ_ASPOR Histone H2A.Z OS=Aspergillus oryzae (strain ATCC 42149 / RIB 40) OX=510516 GN=HTZ1 PE=3 SV=1\nsp|Q0UG93|H2AZ_PHANO Histone H2A.Z OS=Phaeosphaeria nodorum (strain SN15 / ATCC MYA-4574 / FGSC 10173) OX=321614 GN=HTZ1 PE=3 SV=1\nsp|A1D0C1|H2AZ_NEOFI Histone H2A.Z OS=Neosartorya fischeri (strain ATCC 1020 / DSM 3700 / CBS 544.65 / FGSC A1164 / JCM 1740 / NRRL 181 / WB 181) OX=331117 GN=htz1 PE=3 SV=1\nsp|P22647|H2AZ_ONCMY Histone H2A.Z OS=Oncorhynchus mykiss OX=8022 GN=h2az1 PE=1 SV=4\nsp|P08844|H2A_EMENI Histone H2A OS=Emericella nidulans (strain FGSC A4 / ATCC 38163 / CBS 112.46 / NRRL 194 / M139) OX=227321 GN=htaA PE=3 SV=2\nsp|P27325|H2A_URECA Histone H2A OS=Urechis caupo OX=6431 PE=3 SV=2\nsp|Q4PEF9|H2A_USTMA Histone H2A OS=Ustilago maydis (strain 521 / FGSC 9021) OX=237631 GN=HTA1 PE=3 SV=3\nsp|P35066|H2A_TRYCR Histone H2A OS=Trypanosoma cruzi OX=5693 PE=3 SV=2\nsp|Q3ZBX9|H2AJ_BOVIN Histone H2A.J OS=Bos taurus OX=9913 GN=H2AJ PE=2 SV=1\nsp|Q8CGP7|H2A1K_MOUSE Histone H2A type 1-K OS=Mus musculus OX=10090 GN=H2ac15 PE=1 SV=3\nsp|Q1S053|H2A3_MEDTR Probable histone H2A.3 OS=Medicago truncatula OX=3880 PE=3 SV=1\nsp|Q8IUE6|H2A2B_HUMAN Histone H2A type 2-B OS=Homo sapiens OX=9606 GN=H2AC21 PE=1 SV=3\nsp|P0C5Y9|H2AB1_HUMAN Histone H2A-Bbd type 1 OS=Homo sapiens OX=9606 GN=H2AB1 PE=1 SV=1\nsp|P04912|H2A2_YEAST Histone H2A.2 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=HTA2 PE=1 SV=2\nsp|Q6BKW8|H2A1_DEBHA Histone H2A.1 OS=Debaryomyces hansenii (strain ATCC 36239 / CBS 767 / BCRC 21394 / JCM 1990 / NBRC 0083 / IGC 2968) OX=284592 GN=HTA1 PE=3 SV=3\nsp|Q4R3X5|H2AJ_MACFA Histone H2A.J OS=Macaca fascicularis OX=9541 GN=H2AJ PE=2 SV=1\nsp|A2WQG7|H2A5_ORYSI Probable histone H2A.5 OS=Oryza sativa subsp. indica OX=39946 GN=OsI_002060 PE=3 SV=1\nsp|P0C170|H2A1E_RAT Histone H2A type 1-E OS=Rattus norvegicus OX=10116 PE=1 SV=2\nsp|Q64598|H2A1F_RAT Histone H2A type 1-F OS=Rattus norvegicus OX=10116 PE=3 SV=3\nsp|Q9S9K7|H2AXB_ARATH Probable histone H2AXb OS=Arabidopsis thaliana OX=3702 GN=At1g54690 PE=1 SV=1\nsp|O75409|HYPM_HUMAN Huntingtin-interacting protein M OS=Homo sapiens OX=9606 GN=H2AP PE=1 SV=2\nsp|A5DXC6|H2AZ_LODEL Histone H2A.Z OS=Lodderomyces elongisporus (strain ATCC 11503 / CBS 2605 / JCM 1781 / NBRC 1676 / NRRL YB-4239) OX=379508 GN=HTZ1 PE=3 SV=1\nsp|Q9HGX4|H2A_AGABI Histone H2A OS=Agaricus bisporus OX=5341 PE=2 SV=3\nsp|Q5AUJ1|H2AZ_EMENI Histone H2A.Z OS=Emericella nidulans (strain FGSC A4 / ATCC 38163 / CBS 112.46 / NRRL 194 / M139) OX=227321 GN=htz1 PE=3 SV=1\nsp|Q09FM9|H2AZ_MONPU Histone H2A.Z OS=Monascus purpureus OX=5098 GN=HTZ1 PE=3 SV=1\nsp|Q5RC42|H2AZ_PONAB Histone H2A.Z OS=Pongo abelii OX=9601 GN=H2AZ1 PE=2 SV=3\nsp|Q4IMD1|H2AZ_GIBZE Histone H2A.Z OS=Gibberella zeae (strain ATCC MYA-4620 / CBS 123657 / FGSC 9075 / NRRL 31084 / PH-1) OX=229533 GN=HTZ1 PE=3 SV=1\nsp|Q6C341|H2AZ_YARLI Histone H2A.Z OS=Yarrowia lipolytica (strain CLIB 122 / E 150) OX=284591 GN=HTZ1 PE=3 SV=1\nsp|L7HZV6|H2A_MAGOY Histone H2A OS=Magnaporthe oryzae (strain Y34) OX=1143189 GN=HTA1 PE=2 SV=1\nsp|P13630|H2A_PARLI Histone H2A OS=Paracentrotus lividus OX=7656 PE=3 SV=2\nsp|P0DQL2|H2A_SPHLA Buforin-2 OS=Sphaenorhynchus lacteus OX=279984 PE=1 SV=1\nsp|P84057|H2A_TIGCA Histone H2A OS=Tigriopus californicus OX=6832 GN=His2A PE=3 SV=2\nsp|A1A4R1|H2A2C_BOVIN Histone H2A type 2-C OS=Bos taurus OX=9913 GN=H2AC20 PE=2 SV=1\nsp|Q59SU5|H2A1_CANAL Histone H2A.1 OS=Candida albicans (strain SC5314 / ATCC MYA-2876) OX=237561 GN=HTA1 PE=3 SV=3\nsp|P04909|H2A1_SCHPO Histone H2A-alpha OS=Schizosaccharomyces pombe (strain 972 / ATCC 24843) OX=284812 GN=hta1 PE=1 SV=3\nsp|P16865|H2A3_VOLCA Histone H2A-III OS=Volvox carteri OX=3067 PE=3 SV=1\nsp|P02277|H2A3_WHEAT Histone H2A.2.2 OS=Triticum aestivum OX=4565 PE=1 SV=1\nsp|Q54HV9|H2AV2_DICDI Histone H2A.v2 OS=Dictyostelium discoideum OX=44689 GN=H2Av2 PE=3 SV=1\nsp|Q2QPG9|H2AXB_ORYSJ Probable histone H2AXb OS=Oryza sativa subsp. japonica OX=39947 GN=Os12g0530000 PE=2 SV=1\nsp|Q6Y237|H2AV_PAGMA Histone H2A.V OS=Pagrus major OX=143350 GN=h2az2 PE=2 SV=3\nsp|B6MUN4|MYSM1_BRAFL Histone H2A deubiquitinase MYSM1 OS=Branchiostoma floridae OX=7739 GN=MYSM1 PE=3 SV=1\nsp|A0JMR6|MYSM1_XENLA Histone H2A deubiquitinase MYSM1 OS=Xenopus laevis OX=8355 GN=mysm1 PE=2 SV=1\nsp|P69139|H2A3_PSAMI Late histone H2A.3, gonadal OS=Psammechinus miliaris OX=7660 PE=1 SV=2\nsp|Q8LLP5|H2AXA_ORYSJ Probable histone H2AXa OS=Oryza sativa subsp. japonica OX=39947 GN=Os03g0721900 PE=3 SV=1\nsp|Q54WG6|H2AX_DICDI Histone H2AX OS=Dictyostelium discoideum OX=44689 GN=H2AX PE=1 SV=1\nsp|Q6GM86|H2AX_XENLA Histone H2AX OS=Xenopus laevis OX=8355 GN=h2ax PE=1 SV=3\nsp|O04848|H2AXA_ARATH Probable histone H2AXa OS=Arabidopsis thaliana OX=3702 GN=At1g08880 PE=1 SV=1\nsp|Q5F3F2|MYSM1_CHICK Histone H2A deubiquitinase MYSM1 OS=Gallus gallus OX=9031 GN=MYSM1 PE=2 SV=1\nsp|Q5RGA4|MYSM1_DANRE Histone H2A deubiquitinase MYSM1 OS=Danio rerio OX=7955 GN=mysm1 PE=2 SV=2\nsp|C0HKE4|H2A1E_MOUSE Histone H2A type 1-E OS=Mus musculus OX=10090 GN=H2ac8 PE=1 SV=1\nsp|Q8CGP5|H2A1F_MOUSE Histone H2A type 1-F OS=Mus musculus OX=10090 GN=Hist1h2af PE=1 SV=3\nsp|C0HKE5|H2A1G_MOUSE Histone H2A type 1-G OS=Mus musculus OX=10090 GN=H2ac11 PE=1 SV=1\nsp|P09590|H2A3_STRPU Histone H2A-beta, sperm (Fragment) OS=Strongylocentrotus purpuratus OX=7668 PE=1 SV=1\nsp|Q54HV7|H2AV3_DICDI Histone H2A.v3 OS=Dictyostelium discoideum OX=44689 GN=H2Av3 PE=3 SV=1\nsp|Q84MP7|H2AV3_ORYSJ Probable histone H2A variant 3 OS=Oryza sativa subsp. japonica OX=39947 GN=Os03g0743400 PE=2 SV=1\nsp|P35064|H2AX_TETTS Histone H2AX OS=Tetrahymena thermophila (strain SB210) OX=312017 GN=HTA1 PE=1 SV=2\nsp|O75367|H2AY_HUMAN Core histone macro-H2A.1 OS=Homo sapiens OX=9606 GN=MACROH2A1 PE=1 SV=4\nsp|Q6BQE9|H2AZ_DEBHA Histone H2A.Z OS=Debaryomyces hansenii (strain ATCC 36239 / CBS 767 / BCRC 21394 / JCM 1990 / NBRC 0083 / IGC 2968) OX=284592 GN=HTZ1 PE=3 SV=3\nsp|A1CJ10|H2A_ASPCL Histone H2A OS=Aspergillus clavatus (strain ATCC 1007 / CBS 513.65 / DSM 816 / NCTC 3887 / NRRL 1 / QM 1276 / 107) OX=344612 GN=hta1 PE=3 SV=1\nsp|Q1E5N1|H2A_COCIM Histone H2A OS=Coccidioides immitis (strain RS) OX=246410 GN=HTA1 PE=3 SV=1\nsp|P55897|H2A_BUFGR Histone H2A (Fragment) OS=Bufo gargarizans OX=30331 PE=1 SV=1\nsp|Q0CBD2|H2A_ASPTN Histone H2A OS=Aspergillus terreus (strain NIH 2624 / FGSC A1156) OX=341663 GN=hta1 PE=3 SV=1\nsp|P82897|H2A_OLILU Histone H2A (Fragment) OS=Olisthodiscus luteus OX=83000 PE=1 SV=1\nsp|Q6WV67|H2A_MYTTR Histone H2A OS=Mytilus trossulus OX=6551 PE=3 SV=3\nsp|Q96QV6|H2A1A_HUMAN Histone H2A type 1-A OS=Homo sapiens OX=9606 GN=H2AC1 PE=1 SV=3\nsp|Q9C944|H2AV3_ARATH Probable histone H2A variant 3 OS=Arabidopsis thaliana OX=3702 GN=At1g52740 PE=1 SV=1\nsp|Q5Y2C3|H2AY_TETTS Histone H2A.Y OS=Tetrahymena thermophila (strain SB210) OX=312017 GN=HTAY PE=2 SV=2\nsp|A2R702|H2AZ_ASPNC Histone H2A.Z OS=Aspergillus niger (strain CBS 513.88 / FGSC A1513) OX=425011 GN=HTZ1 PE=3 SV=1\nsp|Q2GUP0|H2AZ_CHAGB Histone H2A.Z OS=Chaetomium globosum (strain ATCC 6205 / CBS 148.51 / DSM 1962 / NBRC 6347 / NRRL 1970) OX=306901 GN=HTZ1 PE=3 SV=1\nsp|Q4PHE4|H2AZ_USTMA Histone H2A.Z OS=Ustilago maydis (strain 521 / FGSC 9021) OX=237631 GN=HTZ1 PE=3 SV=1\nsp|P0C0S7|H2AZ_RAT Histone H2A.Z OS=Rattus norvegicus OX=10116 GN=H2az1 PE=1 SV=2\nsp|P48003|H2AZ_SCHPO Histone H2A.Z OS=Schizosaccharomyces pombe (strain 972 / ATCC 24843) OX=284812 GN=pht1 PE=1 SV=3\nsp|P0C0S5|H2AZ_HUMAN Histone H2A.Z OS=Homo sapiens OX=9606 GN=H2AZ1 PE=1 SV=2\nsp|Q12692|H2AZ_YEAST Histone H2A.Z OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=HTZ1 PE=1 SV=3\nsp|P35061|H2A_ACRFO Histone H2A OS=Acropora formosa OX=126732 PE=3 SV=2\nsp|P40280|H2A_MAIZE Histone H2A OS=Zea mays OX=4577 PE=2 SV=1\nsp|P84056|H2A_RHYAM Histone H2A OS=Rhynchosciara americana OX=7186 GN=His2A PE=3 SV=2\nsp|C0HKE1|H2A1B_MOUSE Histone H2A type 1-B OS=Mus musculus OX=10090 GN=H2ac4 PE=1 SV=1\nsp|P20671|H2A1D_HUMAN Histone H2A type 1-D OS=Homo sapiens OX=9606 GN=H2AC7 PE=1 SV=2\nsp|P0C0S8|H2A1_HUMAN Histone H2A type 1 OS=Homo sapiens OX=9606 GN=H2AC11 PE=1 SV=2\nsp|P0C0S9|H2A1_BOVIN Histone H2A type 1 OS=Bos taurus OX=9913 PE=1 SV=2\nsp|Q6ZL42|H2A2_ORYSJ Probable histone H2A.2 OS=Oryza sativa subsp. japonica OX=39947 GN=Os07g0545400 PE=2 SV=1\nsp|A2XZN0|H2A6_ORYSI Probable histone H2A.6 OS=Oryza sativa subsp. indica OX=39946 GN=OsI_017523 PE=3 SV=2\nsp|Q00728|H2A4_RAT Histone H2A type 4 OS=Rattus norvegicus OX=10116 PE=2 SV=2\nsp|A2ZK29|H2A7_ORYSI Probable histone H2A.7 OS=Oryza sativa subsp. indica OX=39946 GN=OsI_036922 PE=3 SV=1\nsp|Q43312|H2A7_WHEAT Protein H2A.7 OS=Triticum aestivum OX=4565 GN=H2A-4 PE=2 SV=3\nsp|P25470|H2A1_PEA Histone H2A.1 OS=Pisum sativum OX=3888 PE=2 SV=1\nsp|Q9SII0|H2AV2_ARATH Probable histone H2A variant 2 OS=Arabidopsis thaliana OX=3702 GN=At2g38810 PE=2 SV=1\nsp|Q6L500|H2A4_ORYSJ Probable histone H2A.4 OS=Oryza sativa subsp. japonica OX=39947 GN=Os05g0461400 PE=2 SV=1\nsp|Q8S857|H2AV2_ORYSJ Probable histone H2A variant 2 OS=Oryza sativa subsp. japonica OX=39947 GN=Os10g0418000 PE=2 SV=1\nsp|Q02874|H2AY_RAT Core histone macro-H2A.1 OS=Rattus norvegicus OX=10116 GN=Macroh2a1 PE=1 SV=4\nsp|Q5AEE1|H2AZ_CANAL Histone H2A.Z OS=Candida albicans (strain SC5314 / ATCC MYA-2876) OX=237561 GN=HTZ1 PE=3 SV=3\nsp|A4QVR2|H2AZ_MAGO7 Histone H2A.Z OS=Magnaporthe oryzae (strain 70-15 / ATCC MYA-4617 / FGSC 8958) OX=242507 GN=HTZ1 PE=3 SV=1\nsp|P21896|H2A_CHITH Histone H2A OS=Chironomus thummi thummi OX=7155 PE=3 SV=2\nsp|Q6PV61|H2A_PENVA Histone H2A OS=Penaeus vannamei OX=6689 PE=1 SV=3\nsp|P84051|H2A_DROME Histone H2A OS=Drosophila melanogaster OX=7227 GN=His2A PE=1 SV=2\nsp|Q6GSS7|H2A2A_MOUSE Histone H2A type 2-A OS=Mus musculus OX=10090 GN=Hist2h2aa1 PE=1 SV=3\nsp|P27891|H2A1_LEIIN Histone H2A.1 OS=Leishmania infantum OX=5671 PE=2 SV=1\nsp|P06897|H2A1_XENLA Histone H2A type 1 OS=Xenopus laevis OX=8355 PE=1 SV=2\nsp|C0HKE9|H2A1P_MOUSE Histone H2A type 1-P OS=Mus musculus OX=10090 GN=Hist1h2ap PE=1 SV=1\nsp|Q9FJE8|H2A7_ARATH Probable histone H2A.7 OS=Arabidopsis thaliana OX=3702 GN=At5g59870 PE=1 SV=1\nsp|Q6ZL43|H2A1_ORYSJ Probable histone H2A.1 OS=Oryza sativa subsp. japonica OX=39947 GN=Os07g0545300 PE=2 SV=1\nsp|O23628|H2AV1_ARATH Histone H2A variant 1 OS=Arabidopsis thaliana OX=3702 GN=H2AV PE=1 SV=1\nsp|Q43208|H2A4_WHEAT Histone H2A.4 OS=Triticum aestivum OX=4565 GN=TH254 PE=2 SV=3\nsp|Q6GM74|H2AV_XENLA Histone H2A.V OS=Xenopus laevis OX=8355 GN=h2az2 PE=2 SV=3\nsp|A2ZK26|H2A8_ORYSI Probable histone H2A.8 OS=Oryza sativa subsp. indica OX=39946 GN=OsI_036919 PE=3 SV=1\nsp|Q3THW5|H2AV_MOUSE Histone H2A.V OS=Mus musculus OX=10090 GN=H2az2 PE=1 SV=3\nsp|O62695|H2AV_RABIT Histone H2A.V OS=Oryctolagus cuniculus OX=9986 GN=H2AZ2 PE=2 SV=3\nsp|P07793|H2A4_PSAMI Late histone H2A.2.2 OS=Psammechinus miliaris OX=7660 PE=3 SV=2\nsp|Q7ZUY3|H2AX_DANRE Histone H2AX OS=Danio rerio OX=7955 GN=h2ax PE=2 SV=3\nsp|Q9QZQ8|H2AY_MOUSE Core histone macro-H2A.1 OS=Mus musculus OX=10090 GN=Macroh2a1 PE=1 SV=3\nsp|A1C5F1|H2AZ_ASPCL Histone H2A.Z OS=Aspergillus clavatus (strain ATCC 1007 / CBS 513.65 / DSM 816 / NCTC 3887 / NRRL 1 / QM 1276 / 107) OX=344612 GN=HTZ1 PE=3 SV=1\nsp|Q54LA5|H2AZ_DICDI Histone H2A.z OS=Dictyostelium discoideum OX=44689 GN=H2AZ PE=3 SV=1\nsp|Q6YNC8|H2AZ_SHEEP Histone H2A.Z OS=Ovis aries OX=9940 GN=H2AZ1 PE=2 SV=3\nsp|P19178|H2A_PLADU Histone H2A OS=Platynereis dumerilii OX=6359 PE=3 SV=2\nsp|Q8X132|H2A_NEUCR Histone H2A OS=Neurospora crassa (strain ATCC 24698 / 74-OR23-1A / CBS 708.71 / DSM 1257 / FGSC 987) OX=367110 GN=hh2a PE=3 SV=3\nsp|P13912|H2A_CAIMO Histone H2A OS=Cairina moschata OX=8855 PE=3 SV=2\nsp|P02269|H2A_ASTRU Histone H2A OS=Asterias rubens OX=7604 PE=1 SV=2\nsp|Q6C4I6|H2A_YARLI Histone H2A OS=Yarrowia lipolytica (strain CLIB 122 / E 150) OX=284591 GN=HTA1 PE=3 SV=3\nsp|P84053|H2A_DROHY Histone H2A OS=Drosophila hydei OX=7224 GN=His2A PE=3 SV=2\nsp|P59890|H2A_HIPHI Histone H2A (Fragment) OS=Hippoglossus hippoglossus OX=8267 PE=1 SV=2\nsp|P04735|H2A1_PSAMI Late histone H2A.1 OS=Psammechinus miliaris OX=7660 PE=2 SV=2\nsp|A5DJJ2|H2A2_PICGU Histone H2A.2 OS=Meyerozyma guilliermondii (strain ATCC 6260 / CBS 566 / DSM 6381 / JCM 1539 / NBRC 10279 / NRRL Y-324) OX=294746 GN=HTA2 PE=3 SV=1\nsp|Q4FZT6|H2A3_RAT Histone H2A type 3 OS=Rattus norvegicus OX=10116 PE=2 SV=3\nsp|Q75CC6|H2AZ_ASHGO Histone H2A.Z OS=Ashbya gossypii (strain ATCC 10895 / CBS 109.51 / FGSC 9923 / NRRL Y-1056) OX=284811 GN=HTZ1 PE=3 SV=3\nsp|Q4WE68|H2AZ_ASPFU Histone H2A.Z OS=Neosartorya fumigata (strain ATCC MYA-4609 / Af293 / CBS 101355 / FGSC A1100) OX=330879 GN=htz1 PE=3 SV=1\nsp|Q5ZMD6|H2AZ_CHICK Histone H2A.Z OS=Gallus gallus OX=9031 GN=H2AFZ PE=1 SV=3\nsp|A3GHC1|H2AZ_PICST Histone H2A.Z OS=Scheffersomyces stipitis (strain ATCC 58785 / CBS 6054 / NBRC 10063 / NRRL Y-11545) OX=322104 GN=HTZ1 PE=3 SV=2\nsp|P0CO00|H2AZ_CRYNJ Histone H2A.Z OS=Cryptococcus neoformans var. neoformans serotype D (strain JEC21 / ATCC MYA-565) OX=214684 GN=HTZ1 PE=3 SV=1\nsp|P0C953|H2A_ASPNC Histone H2A OS=Aspergillus niger (strain CBS 513.88 / FGSC A1513) OX=425011 GN=httA PE=3 SV=1\nsp|Q6CUC8|H2AZ_KLULA Histone H2A.Z OS=Kluyveromyces lactis (strain ATCC 8585 / CBS 2359 / DSM 70799 / NBRC 1267 / NRRL Y-1140 / WM37) OX=284590 GN=HTZ1 PE=3 SV=3\nsp|Q6CK59|H2A_KLULA Histone H2A OS=Kluyveromyces lactis (strain ATCC 8585 / CBS 2359 / DSM 70799 / NBRC 1267 / NRRL Y-1140 / WM37) OX=284590 GN=HTA1 PE=1 SV=3\nsp|P0CN98|H2A_CRYNJ Histone H2A OS=Cryptococcus neoformans var. neoformans serotype D (strain JEC21 / ATCC MYA-565) OX=214684 GN=HTA1 PE=3 SV=1\nsp|Q2U5A8|H2A_ASPOR Histone H2A OS=Aspergillus oryzae (strain ATCC 42149 / RIB 40) OX=510516 GN=hta1 PE=3 SV=3\nsp|Q8SSG3|H2A_ENCCU Histone H2A OS=Encephalitozoon cuniculi (strain GB-M1) OX=284813 GN=HTA1 PE=1 SV=1\nsp|Q757L4|H2A2_ASHGO Histone H2A.2 OS=Ashbya gossypii (strain ATCC 10895 / CBS 109.51 / FGSC 9923 / NRRL Y-1056) OX=284811 GN=HTA2 PE=3 SV=3\nsp|P25469|H2A1_SOLLC Histone H2A.1 OS=Solanum lycopersicum OX=4081 PE=2 SV=1\nsp|Q9LD75|H2A2_LILLO Histone H2A.2 OS=Lilium longiflorum OX=4690 GN=gH2A PE=1 SV=1\nsp|P06898|H2A2_XENLA Histone H2A type 2 OS=Xenopus laevis OX=8355 PE=3 SV=3\nsp|Q07135|H2AO_CHITH Histone H2A, orphon OS=Chironomus thummi thummi OX=7155 PE=3 SV=2\nsp|Q2HU68|H2A1_MEDTR Probable histone H2A.1 OS=Medicago truncatula OX=3880 GN=MtrDRAFT_AC149210g1v1 PE=3 SV=1\nsp|Q9LHQ5|H2A2_ARATH Probable histone H2A.2 OS=Arabidopsis thaliana OX=3702 GN=At3g20670 PE=1 SV=1\nsp|A2YVE5|H2A3_ORYSI Probable histone H2A.3 OS=Oryza sativa subsp. indica OX=39946 GN=OsI_028288 PE=3 SV=1\nsp|Q2QS71|H2A7_ORYSJ Probable histone H2A.7 OS=Oryza sativa subsp. japonica OX=39947 GN=Os12g0438000 PE=2 SV=1\nsp|Q54HW2|H2AV1_DICDI Histone H2A.v1 OS=Dictyostelium discoideum OX=44689 GN=H2Av1 PE=3 SV=1\nsp|P08992|H2AV_TETTS Histone H2A.Z OS=Tetrahymena thermophila (strain SB210) OX=312017 GN=HTA3 PE=1 SV=2\nsp|Q8CGP6|H2A1H_MOUSE Histone H2A type 1-H OS=Mus musculus OX=10090 GN=H2ac12 PE=1 SV=3\nsp|Q5BJ65|H2AV_XENTR Histone H2A.V OS=Xenopus tropicalis OX=8364 GN=h2az2 PE=2 SV=3\nsp|Q94E96|H2A5_ORYSJ Probable histone H2A.5 OS=Oryza sativa subsp. japonica OX=39947 GN=Os01g0502700 PE=2 SV=1\nsp|A3LZZ0|H2A1_PICST Histone H2A.1 OS=Scheffersomyces stipitis (strain ATCC 58785 / CBS 6054 / NBRC 10063 / NRRL Y-11545) OX=322104 GN=HTA2 PE=3 SV=1\nsp|P40281|H2A2_PEA Histone H2A.2 OS=Pisum sativum OX=3888 PE=2 SV=1\nsp|C0HKE8|H2A1O_MOUSE Histone H2A type 1-O OS=Mus musculus OX=10090 GN=Hist1h2ao PE=1 SV=1\nsp|A3LXE7|H2A2_PICST Histone H2A.2 OS=Scheffersomyces stipitis (strain ATCC 58785 / CBS 6054 / NBRC 10063 / NRRL Y-11545) OX=322104 GN=HTA2 PE=3 SV=1\nsp|Q75L11|H2A6_ORYSJ Probable histone H2A.6 OS=Oryza sativa subsp. japonica OX=39947 GN=Os05g0113900 PE=2 SV=2\nsp|P02276|H2A2_WHEAT Histone H2A.2.1 OS=Triticum aestivum OX=4565 PE=1 SV=1\nsp|A5DWF1|H2A1_LODEL Histone H2A.1 OS=Lodderomyces elongisporus (strain ATCC 11503 / CBS 2605 / JCM 1781 / NBRC 1676 / NRRL YB-4239) OX=379508 GN=HTA1 PE=3 SV=1\nsp|A2XLI0|H2AXA_ORYSI Probable histone H2AXa OS=Oryza sativa subsp. indica OX=39946 GN=OsI_012923 PE=3 SV=1\nsp|Q6DER6|H2AZL_XENTR Histone H2A.Z-like OS=Xenopus tropicalis OX=8364 GN=h2a.zl PE=2 SV=3\nsp|Q0C919|H2AZ_ASPTN Histone H2A.Z OS=Aspergillus terreus (strain NIH 2624 / FGSC A1156) OX=341663 GN=HTZ1 PE=3 SV=1\nsp|Q1DTG2|H2AZ_COCIM Histone H2A.Z OS=Coccidioides immitis (strain RS) OX=246410 GN=HTZ1 PE=3 SV=1\nsp|P0C0S6|H2AZ_MOUSE Histone H2A.Z OS=Mus musculus OX=10090 GN=H2az1 PE=1 SV=2\nsp|P84055|H2A_DROYA Histone H2A OS=Drosophila yakuba OX=7245 GN=His2A PE=2 SV=2\nsp|Q873G4|H2AZ_NEUCR Histone H2A.Z OS=Neurospora crassa (strain ATCC 24698 / 74-OR23-1A / CBS 708.71 / DSM 1257 / FGSC 987) OX=367110 GN=hH2Az PE=3 SV=1\nsp|P0CT12|H2A_MAGO7 Histone H2A OS=Magnaporthe oryzae (strain 70-15 / ATCC MYA-4617 / FGSC 8958) OX=242507 GN=HTA1 PE=3 SV=1\nsp|C0HJQ3|H2A_ACIGU Histone H2A (Fragments) OS=Acipenser gueldenstaedtii OX=7902 PE=1 SV=1\nsp|P02270|H2A_SIPNU Histone H2A OS=Sipunculus nudus OX=6446 PE=1 SV=2\nsp|P50567|H2A_CHLRE Histone H2A OS=Chlamydomonas reinhardtii OX=3055 GN=H2A-II PE=3 SV=1\nsp|Q6WV88|H2A_MYTGA Histone H2A OS=Mytilus galloprovincialis OX=29158 PE=3 SV=3\nsp|P84054|H2A_DROSI Histone H2A OS=Drosophila simulans OX=7240 GN=His2A PE=3 SV=2\nsp|Q9XG56|H2A1_LILLO Histone H2A.1 OS=Lilium longiflorum OX=4690 GN=gcH2A PE=2 SV=1\nsp|P04910|H2A2_SCHPO Histone H2A-beta OS=Schizosaccharomyces pombe (strain 972 / ATCC 24843) OX=284812 GN=hta2 PE=1 SV=2\nsp|Q43214|H2A6_WHEAT Protein H2A.6 OS=Triticum aestivum OX=4565 GN=H2A-3 PE=2 SV=3\nsp|Q6FWM7|H2A1_CANGA Histone H2A.1 OS=Candida glabrata (strain ATCC 2001 / CBS 138 / JCM 3761 / NBRC 0622 / NRRL Y-65) OX=284593 GN=HTA1 PE=3 SV=3\nsp|Q64523|H2A2C_MOUSE Histone H2A type 2-C OS=Mus musculus OX=10090 GN=H2ac20 PE=1 SV=3\nsp|Q96KK5|H2A1H_HUMAN Histone H2A type 1-H OS=Homo sapiens OX=9606 GN=H2AC12 PE=1 SV=3\nsp|C0HKE7|H2A1N_MOUSE Histone H2A type 1-N OS=Mus musculus OX=10090 GN=Hist1h2an PE=1 SV=1\nsp|Q71UI9|H2AV_HUMAN Histone H2A.V OS=Homo sapiens OX=9606 GN=H2AZ2 PE=1 SV=3\nsp|Q9LD28|H2A6_ARATH Histone H2A.6 OS=Arabidopsis thaliana OX=3702 GN=RAT5 PE=1 SV=1\nsp|P35063|H2AX_PICAB Histone H2AX OS=Picea abies OX=3329 PE=2 SV=1\nsp|Q9CR04|HYPM_MOUSE Huntingtin-interacting protein M OS=Mus musculus OX=10090 GN=H2ap PE=3 SV=1\nsp|A1D8G8|H2A_NEOFI Histone H2A OS=Neosartorya fischeri (strain ATCC 1020 / DSM 3700 / CBS 544.65 / FGSC A1164 / JCM 1740 / NRRL 181 / WB 181) OX=331117 GN=hta1 PE=3 SV=1\nsp|Q6WV66|H2A_MYTCA Histone H2A OS=Mytilus californianus OX=6549 PE=3 SV=3\nsp|Q6WV69|H2A_MYTCH Histone H2A OS=Mytilus chilensis OX=173667 PE=3 SV=3\nsp|P02268|H2A_SEPOF Histone H2A OS=Sepia officinalis OX=6610 PE=1 SV=2\nsp|P69140|H2A_PARAN Histone H2A, gonadal OS=Parechinus angulosus OX=7658 PE=1 SV=2\nsp|P19177|H2A_PETCR Histone H2A OS=Petroselinum crispum OX=4043 PE=2 SV=1\nsp|P69141|H2A_STRPU Histone H2A, embryonic OS=Strongylocentrotus purpuratus OX=7668 PE=1 SV=2\nsp|P0CC09|H2A2A_RAT Histone H2A type 2-A OS=Rattus norvegicus OX=10116 GN=H2ac18 PE=1 SV=1\nsp|Q7L7L0|H2A3_HUMAN Histone H2A type 3 OS=Homo sapiens OX=9606 GN=H2AW PE=1 SV=3\nsp|P09589|H2A3_LYTPI Histone H2A, sperm (Fragment) OS=Lytechinus pictus OX=7653 PE=2 SV=1\nsp|P16886|H2AL_STRPU Late histone H2A.L3 OS=Strongylocentrotus purpuratus OX=7668 PE=1 SV=2\nsp|Q71PD7|H2AV_DANRE Histone H2A.V OS=Danio rerio OX=7955 GN=h2az2a PE=2 SV=3\nsp|Q8H7Y8|H2AV1_ORYSJ Probable histone H2A variant 1 OS=Oryza sativa subsp. japonica OX=39947 GN=Os03g0162200 PE=2 SV=1\nsp|P08985|H2AV_DROME Histone H2A.v OS=Drosophila melanogaster OX=7227 GN=His2Av PE=1 SV=2\nsp|P08991|H2AV_STRPU Histone H2A.V (Fragment) OS=Strongylocentrotus purpuratus OX=7668 GN=H2A.F/Z PE=1 SV=1\n\nWe found 300 records using the search term: H2A\n","output_type":"stream"}]},{"cell_type":"markdown","source":"***\n\n<font color=blue><b>STEP 9:</b></font> Edit the code above to search the description rather than the ID. Make sure to change it in both places!\n\n***\n\nAs you have seen above, the IDs are a little long and redundant with the name. The code below simplifies the record ID and writes a new, simpler file.\n\n<font color=blue><b>STEP 10:</b></font> Read through the code below and run it. \n\nChange the name of the original file to your FASTA file <b>(files/uniprot_histone.fasta)</b>.\n\nChange the name of the simplified file to <b>\"uniprot_histone_simple.fasta\"</b>.","metadata":{}},{"cell_type":"code","source":"original_file = \"<<<your filename and path>>>\"         # original file path\nsimple_file = \"<<<new filename and path>>>\"   # new file path\n\nwith open(original_file) as original, open(simple_file, 'w') as simple:  #opens the file to read and one to write\n    records = SeqIO.parse(original, 'fasta')   # here we use the Biopython SeqIO again to parse the file into records\n\n    for item in records:                       # iterate through each item in the list called records\n        tmp_name = str.split(item.id, \"|\")[1]      # sets the variable tmp_name to the second item generated by splitting at the | character \n        item.id = tmp_name                         # now this sets the id of that item to the name\n        SeqIO.write(item, simple, 'fasta')  # writes out the item information to the new, corrected fasta file","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"Let's consider how we changed this file by looking at the simplfied file.\n\n<font color=blue><b>STEP 11:</b></font> You can find the new file in the file browser in the left of your screen and double click to open it.","metadata":{}},{"cell_type":"markdown","source":"***\n## Big Data Strategies - Filtering and Reducing Redundancy in Datasets\n\nWe still have a lot of sequences to deal with. It is often best to use fast and easy methods to pare down a dataset before using more computationally intensive ones. \n\nWe will start by filtering out small fragments and large proteins, then move on to removing redundancy.\n***","metadata":{}},{"cell_type":"markdown","source":"## Filtering by length\n\nLet's visualize the distribution of sequence lengths in our dataset. The code below will generate a histogram of the lengths of sequences. \n\n<font color=blue><b>STEP 12:</b></font> Run the code below to generate a histogram of sequence lengths. \n\nBe sure to updat the name of the input file to <b>\"uniprot_histone_simple.fasta\"</b>.\n","metadata":{}},{"cell_type":"code","source":"import matplotlib.pyplot as plt\n\nrecords = list(SeqIO.parse(\"<<<your filename and path>>>\", \"fasta\"))\n\nlengths = []\n\nfor i in records:\n    #print(i.seq)\n    lengths.append(len(i.seq))\n\nlengths.sort()\n#print(lengths)\n\nplt.hist(lengths, bins = 100)\nplt.xlabel('seq length')\nplt.ylabel('count')\nplt.show()","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"<font color=blue><b>STEP 13:</b></font> Answer the question:\n\n    1. Using the histogram, what are the most common lengths of proteins in this dataset?\n\n***\n\nWe will further reduce the complexity of this dataset by removing sequences that are much larger and much smaller than the most common lengths shown. Those variables can be determined using a histogram and easily changed for different datasets. Using the histogram we can identify values for the small_len=100 and large_len=400 variables.\n\n<font color=blue><b>STEP 14:</b></font>   Run the code below to remove short and long sequences and generate a new fasta file called <b>\"files/uniprot_histone_simple_trim.fasta\"</b>.\n","metadata":{}},{"cell_type":"code","source":"from Bio import SeqIO\n\noriginal_file = \"<<filename and path>>\"\ntrimmed_file = \"<<new filename and path>>\"\n\nsmall_len = 100\nlarge_len = 400\n\nwith open(original_file) as original, open(trimmed_file, 'w') as trimmed:\n    records = SeqIO.parse(original_file, 'fasta')\n\n    for record in records:\n        if len(record.seq) > small_len and len(record.seq) < large_len: \n            #print(len(record.seq))\n            SeqIO.write(record, trimmed, 'fasta')\n            \nrecords2 = list(SeqIO.parse(\"<<new filename and path>>\", \"fasta\"))\nprint(\"We now have %i sequences in our fasta file.\" % len(records2))","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"<b>How many sequences are in your FASTA file now??</b>\n***\n## Reducing sequence redundancy using CD-HIT\n\nWe will use the program CD-HIT to remove sequences within a given sequence similarity to another sequence. We will use the flag \"-c 0.5\" to keep only sequences with less than 50% sequence identity. The -n flag selects the word size used in processing. Less sequence identity requires the use of smaller words in comparing the sequences. From the CD-HIT manual:\n\n    Choice'of'word'size:\n        -n 5 for thresholds 0.7 ~ 1.0\n        -n 4 for thresholds 0.6 ~ 0.7\n        -n 3 for thresholds 0.5 ~ 0.6\n        -n 2 for thresholds 0.4 ~ 0.5\n\nFor this exercise we will use a cutoff of 50%. <b>In practice you might select higher values, but this will increase runtimes for the following steps and the visualization in the next Jupyter Notebook.</b>\n\n<font color=blue><b>STEP 15:</b></font> Let's run the code below. This runs CD-HIT using the simplified and trimmed fasta as input (-i) and creates a new fasta output (-o) named <b>uniprot_histone_50.fasta</b> that will be much smaller. The bang (!) at the beginning tells the notebook this is not a Python command, but rather a program to be run from the unix environment. <b>Note running CD-HIT takes a few minutes! ","metadata":{}},{"cell_type":"code","source":"!cd-hit -i files/<<filename and path>> -o files/<<new filenam and path>> -c 0.5 -n 3","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"CD-HIT is one strategy for dealing with very large datasets - they can be reduced. However, reduction needs to be done in a logical and reproducible manner. CD-HIT does just that.","metadata":{}},{"cell_type":"markdown","source":"***\n## Adding in sequences of known function\n\nWe are almost done processing our FASTA dataset. In order to ensure that we can identify our histone sequences, we will add a small set of reference FASTA sequences. We will add these knowns to the FASTA file using the cat command (it stands for concatenate). \n\n<font color=blue><b>STEP 16:</b></font> Run the code below which combines the two files into a new file called \"files/final_50.fasta.","metadata":{}},{"cell_type":"code","source":"!cat \"<<your filename and path>>\" \"files/known_histone.fasta\" > \"files/final_50.fasta\"","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"<font color=blue><b>STEP 17:</b></font> Run the code below to determine the final number of sequences in your file! ","metadata":{}},{"cell_type":"code","source":"records = list(SeqIO.parse(\"files/final_50.fasta\", \"fasta\"))    # use SeqIO to process the file\n\nprint(\"There are %i sequences in your file.\\n\" % len(records))  # print the number of sequences","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"***\n\n## Making a BLAST-able database and performing an all-by-all BLAST search\n\nNow that we have our dataset we are going to use NCBI BLAST tools to 1) create a searchable database using our processed set of sequences and 2) use protein BLAST to complete an all-by-all search of that database using each of those sequences.\n\nWe will perform an analysis of the evolutionary links between these sequences by determining which sequences are most closely related to which others.\n\n","metadata":{}},{"cell_type":"markdown","source":"<font color=blue><b>STEP 18:</b></font> In the code box below we have added input and output files and told the program it is a protein database. The input is the files/final_40.fasta file. We will name the output files files/finalpro_40. Go ahead and run this code.","metadata":{}},{"cell_type":"code","source":"!makeblastdb -in files/final_50.fasta -dbtype prot -out files/finalpro_50","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"<font color=blue><b>STEP 19:</b></font> Next we want to use BLASTp (protein query vs protein database BLAST) to take each of our protein sequences and search the database for connections. Let's start by finding out the usage of blastp by running the command below.","metadata":{}},{"cell_type":"markdown","source":"<font color=blue><b>STEP 20:</b></font> Run the code below to use blastp to search the files/finalpro_50 database using each of the sequences in the files/final_50.fasta file.\n\nThe outfmt controls the output formatting and the -evalue is the expectation value. Smaller evalues help us to ensure that the results are not found simply by chance. In the code below we are using a value of 10e-40.","metadata":{}},{"cell_type":"code","source":"!blastp -db files/finalpro_50 -query files/final_50.fasta -outfmt \"6 qseqid sseqid evalue\" -out files/BLASTe40_out -num_threads 4 -evalue 10e-40","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"Once the above code has completed running you should see a file in the file browswer at the left called \"BLASTe40_out\". This is simply a text file, but contains the information we want to visualize in the next Jupyter Notebook. <b>Let's download \"BLASTe40_out\" and \"final_40.fasta\" to our personal computers so we can use them later.<b> \n    \n<font color=blue><b>STEP 21:</b></font> Download the file by clicking on the file \"BLASTe40_out\" and then go up to the Jupyter file menu and Download. Repeat for \"final_40.fasta\" Please note where you are downloading these files on your computer.\n","metadata":{}},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[]}]}